src/app/BufferedReadCallback.cpp

```diff
@@ -247,13 +249,14 @@ void BufferedReadCallback::OnAttributeData(const ReadClient * apReadClient, cons
     }
     else
     {
-        mCallback.OnAttributeData(apReadClient, aPath, apData, aStatus);
+        mCallback.OnAttributeData(apReadClient, apVersion, aPath, apData, aStatus);
     }
 
     //
     // Update our latched buffered path.
     //
     mBufferedPath = aPath;
+    mpDataVersion = apVersion;
```

Comment:
I don't understand how that can possibly be safe.  We're storing a pointer to ... where?  How long will the memory it points to live?

Reply:
just update with optional<DataVersion>